### ISSUE TITLE
Snapper root config fails to create snapshot on Arch

### ISSUE THREAD

**Issue by Code-Ac** [openSUSE/snapper#693]
Hello, I am currently trying to add snapper as my backup system. I ran `sudo snapper create-config /` and that worked but now I get errors when trying to create the first snapshot whit `sudo snapper -c root create -d "init"`. The command returns "Creating snapshot failed.", logs: 
```
2022-02-26 20:41:01 WAR libsnapper(17992) FileUtils.cc(SDir):91 - THROW: open failed path://.snapshots/1 errno:2 (No such file or directory)
2022-02-26 20:41:01 WAR libsnapper(17992) Btrfs.cc(checkSnapshot):481 - CAUGHT: open failed path://.snapshots/1 errno:2 (No such file or directory)
2022-02-26 20:41:01 ERR libsnapper(17992) Btrfs.cc(createSnapshot):327 - create snapshot failed, ioctl(BTRFS_IOC_SNAP_CREATE_V2) failed, errno:26 (Text file busy)
2022-02-26 20:41:01 WAR libsnapper(17992) Btrfs.cc(createSnapshot):328 - THROW: create snapshot failed
2022-02-26 20:41:01 WAR libsnapper(17992) Snapshot.cc(createHelper):680 - CAUGHT: create snapshot failed
2022-02-26 20:41:01 WAR libsnapper(17992) Snapshot.cc(createHelper):685 - RETHROW: create snapshot failed
2022-02-26 20:41:01 WAR libsnapper(17992) Client.cc(dispatch):1798 - CAUGHT: create snapshot failed
```

After that, I tried editing the root config option SUBVOLUME from "/" to "" I get "IO Error (open failed path: errno:2 (No such file or directory)).", logs:
```
2022-02-26 21:13:57 MIL libsnapper(136238) snapperd.cc(main):278 - Requesting DBus name
2022-02-26 21:13:57 MIL libsnapper(136238) snapperd.cc(main):293 - Loading snapper configs
2022-02-26 21:13:57 MIL libsnapper(136238) Snapper.cc(getConfigs):290 - Snapper get-configs
2022-02-26 21:13:57 MIL libsnapper(136238) Snapper.cc(getConfigs):291 - libsnapper version 0.9.1
2022-02-26 21:13:57 MIL libsnapper(136238) AsciiFile.cc(reload):115 - loading file /etc/conf.d/snapper
2022-02-26 21:13:57 MIL libsnapper(136238) AsciiFile.cc(getValue):242 - key:SNAPPER_CONFIGS value:root
2022-02-26 21:13:57 MIL libsnapper(136238) AsciiFile.cc(reload):115 - loading file /etc/snapper/configs/root
2022-02-26 21:13:57 MIL libsnapper(136238) AsciiFile.cc(getValue):242 - key:SUBVOLUME value:
2022-02-26 21:13:57 MIL libsnapper(136238) AsciiFile.cc(getValue):242 - key:ALLOW_USERS value:
2022-02-26 21:13:57 MIL libsnapper(136238) AsciiFile.cc(getValue):242 - key:ALLOW_GROUPS value:
2022-02-26 21:13:57 MIL libsnapper(136238) snapperd.cc(main):306 - Listening for method calls and signals
2022-02-26 21:13:57 MIL libsnapper(136238) Snapper.cc(Snapper):92 - Snapper constructor
2022-02-26 21:13:57 MIL libsnapper(136238) Snapper.cc(Snapper):93 - libsnapper version 0.9.1
2022-02-26 21:13:57 MIL libsnapper(136238) Snapper.cc(Snapper):94 - config_name:root disable_filters:false
2022-02-26 21:13:57 MIL libsnapper(136238) AsciiFile.cc(reload):115 - loading file /etc/snapper/configs/root
2022-02-26 21:13:57 MIL libsnapper(136238) AsciiFile.cc(getValue):242 - key:SUBVOLUME value:
2022-02-26 21:13:57 MIL libsnapper(136238) AsciiFile.cc(getValue):242 - key:FSTYPE value:btrfs
2022-02-26 21:13:57 MIL libsnapper(136238) AsciiFile.cc(getValue):242 - key:QGROUP value:
2022-02-26 21:13:57 MIL libsnapper(136238) AsciiFile.cc(getValue):242 - key:SYNC_ACL value:no
2022-02-26 21:13:57 MIL libsnapper(136238) Snapper.cc(Snapper):125 - subvolume: filesystem:btrfs
2022-02-26 21:13:57 MIL libsnapper(136238) Snapper.cc(loadIgnorePatterns):195 - number of ignore patterns:8
2022-02-26 21:13:57 WAR libsnapper(136238) FileUtils.cc(SDir):66 - THROW: open failed path: errno:2 (No such file or directory)
2022-02-26 21:13:57 ERR libsnapper(136238) Snapshot.cc(initialize):364 - reading failed
2022-02-26 21:13:57 WAR libsnapper(136238) FileUtils.cc(SDir):66 - THROW: open failed path: errno:2 (No such file or directory)
2022-02-26 21:13:57 WAR libsnapper(136238) Client.cc(dispatch):1828 - CAUGHT: open failed path: errno:2 (No such file or directory)
```
I think the config file is cached in the upper log.

I am running Arch Linux, using btrfs as fs and tried both the snapper package as the snapper-git package (compile from source)
snapper-git is version 0.9.1.

**Comment by aschnell** [comment]
Maybe you have a swapfile? See http://snapper.io/troubleshooting.html.

**Comment by Code-Ac** [comment]
Ok yes I got one, even tho it’s not that error message.
Will try that and come back later.

**Comment by Code-Ac** [comment]
Yeah after fiddling around whit data compression and copy-on-write my swapfile is now in its own subvolume and everything works!
Thanks for the quick help.

**Comment by tetoNidan** [comment]
I ran into this issue. It was because I built a swap file in the root directory that was trying to snapshotted.

From the [BTRFS Doc's on hibernation](https://btrfs.readthedocs.io/en/latest/ch-swapfile.html)

`subvolume - cannot be snapshotted if it contains any active swapfiles`

If you shut down the swap `sudo swapoff /swap/swapfile` snap-pac works as expected.

Hopefully this helps the next person.
